pkg/summarizer/summary.go

```diff
@@ -486,53 +488,61 @@ func overallStatus(grid *statepb.Grid, recent int, stale string, alerts []*summa
 	return summarypb.DashboardTabSummary_UNKNOWN
 }
 
-func fmtStatus(passCols, cols, passCells, cells int) string {
-	colCent := 100 * float64(passCols) / float64(cols)
-	cellCent := 100 * float64(passCells) / float64(cells)
-	return fmt.Sprintf("%d of %d (%.1f%%) recent columns passed (%d of %d or %.1f%% cells)", passCols, cols, colCent, passCells, cells, cellCent)
-}
-
-func statusMessage(cols int, rows []*statepb.Row, recent int) string {
-	//  2483 of 115784 tests (2.1%) and 163 of 164 runs (99.4%) failed in the past 7 days
+// Culminate set of metrics related to a section of the Grid that can be consumed by
+func gridMetrics(cols int, rows []*statepb.Row, recent int) (int, int, int, int, []float32) {
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 
 	results := results(ctx, rows)
+	var columnFailureRatios []float32
 	var passingCells int
 	var filledCells int
 	var passingCols int
 	var completedCols int
-	for idx := 0; idx < cols; idx++ {
-		if idx >= recent {
-			break
-		}
-		var passes bool
-		var failures bool
+
+	for idx := 0; idx < cols && idx < recent; idx++ {
+		var passes int
+		var failures int
 		for _, ch := range results {
 			// TODO(fejta): fail old running cols
 			switch coalesceResult(<-ch, result.IgnoreRunning) {
-			case statepb.Row_PASS:
-				if !failures {
-					passes = true
-				}
+			case statepb.Row_PASS, statepb.Row_PASS_WITH_ERRORS, statepb.Row_PASS_WITH_SKIPS:
```

Comment:
Can you add a TODO (feel free to assign to me) to make 'what's a passing status' and 'what's a failing status' a common utility function?

We should probably open-source status.go to make sure everyone's consistent about these kinds of concepts, since it gets tricky fast...

Reply:
I agree thanks, I am happy to try to do it as well!